bitswap/network/httpnet/httpnet.go

```diff
@@ -388,6 +388,19 @@ func (ht *Network) Self() peer.ID {
 	return ht.host.ID()
 }
 
+func connectError(errs map[*network.ParsedURL]error) error {
+	if len(errs) == 0 {
+		return ErrNoHTTPAddresses
+	}
+
+	finalErr := ErrNoSuccess
+	for u, err := range errs {
+		finalErr = fmt.Errorf("%w\n* [%s: %s]", finalErr, u.Multiaddress.String(), err)
+	}
```

Comment:
Consider using the [`go.uber.org/multierr`](https://pkg.go.dev/go.uber.org/multierr) package here, unless there is a specific reason to recursively wrap the errors here.

Seems like [`multierr.Combine`](https://pkg.go.dev/go.uber.org/multierr#Combine) might be good here.